File: src/main/groovy/net/zomis/brainf/model/classic/BFLoop.java
package net.zomis.brainf.model.classic;

import net.zomis.brainf.model.BrainfuckCommand;
import net.zomis.brainf.model.BrainfuckRunner;

@Deprecated
public class BFLoop implements BrainfuckCommand {

    private final int i;

    public BFLoop(int i) {
        this.i = i;
    }


    @Override
    public void perform(BrainfuckRunner runner) {
        if (i == 1) {
            if (runner.getMemory().getValue() == 0) {
                throw new UnsupportedOperationException();
            }
        } else if (i == -1) {
            if (runner.getMemory().getValue() != 0) {
                throw new UnsupportedOperationException();
            }
        } else {
            throw new IllegalStateException("Invalid i: " + i);
        }
    }
}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/classic/BFLoop.kt
package net.zomis.brainf.model.classic;

import net.zomis.brainf.model.BrainfuckCommand;
import net.zomis.brainf.model.BrainfuckRunner;

@Deprecated("")
class BFLoop(val i: Int) : BrainfuckCommand {

    override fun perform(runner: BrainfuckRunner) {
        if (i == 1) {
            if (runner.memory.value == 0) {
                throw UnsupportedOperationException();
            }
        } else if (i == -1) {
            if (runner.memory.value != 0) {
                throw UnsupportedOperationException();
            }
        } else {
            throw IllegalStateException("Invalid i: $i");
        }
    }
}
